timesource/timesource.go

```diff
@@ -217,23 +219,28 @@ func (s *NTPTimeSource) Start() {
 		logutils.ZapLogger().Error("failed to update offset", zap.Error(err))
 	}
 
-	s.runPeriodically(s.updateOffset, err == nil)
+	s.runPeriodically(ctx, s.updateOffset, err == nil)
 
 	s.started = true
+	s.cancel = cancel
+
+	return nil
 }
 
 // Stop goroutine that updates time source.
-func (s *NTPTimeSource) Stop() error {
-	if s.quit == nil {
-		return nil
+func (s *NTPTimeSource) Stop() {
+	if s.cancel == nil {
+		return
 	}
-	close(s.quit)
+	s.cancel()
 	s.started = false
-	return nil
 }
 
 func (s *NTPTimeSource) GetCurrentTime() time.Time {
-	s.Start()
+	err := s.Start(context.Background())
+	if err != nil {
+		panic("could not obtain timesource")
```

Comment:
To bring some more detail about the generated error. The syntax is not correct though ;P
```suggestion
		panic("could not obtain timesource in timesource.go GetCurrentTime: ", err)
```

Reply:
ooh, the panic should for sure add the file and line number to know where the error is coming from, but let's for sure add the error message :))